src/Symfony/Component/HttpFoundation/File/UploadedFile.php

```diff
@@ -79,6 +79,19 @@ public function getClientOriginalName()
         return $this->originalName;
     }
 
+    /**
+     * Returns the original file name without extension.
+     *
+     * It is extracted from the original file name that was uploaded.
+     * Then it should not be considered as a safe value.
+     *
+     * @return string The origin name without extension
+     */
+    public function getClientOriginalNameWithoutExtension()
+    {
+        return pathinfo($this->originalName, PATHINFO_FILENAME);
```

Comment:
I am afraid [using pathinfo unsafe for utf-8 filenames](https://stackoverflow.com/questions/4451664/make-php-pathinfo-return-the-correct-filename-if-the-filename-is-utf-8)

Reply:
The topic you have mentioned is 7 years old and the origin bug was fixed a long time ago.  Since Symfony require php7 - we should not worry about PHP 5.1 bugs ;)

Reply:
fell free check by yourself https://3v4l.org/ejZCL . It works fine only on HHVM 😄

Reply:
In any case - tests that covers new functionality is a must. Let's add some tests that covers new method and after then check.

Reply:
I wish not to trust any online sandboxes. On my system, it works as expected. https://gist.github.com/arku31/775c5d7839c21feb9003d3014f9b49b8